torchdata/datapipes/iter/util/header.py

```diff
@@ -53,3 +56,65 @@ def __len__(self) -> int:
                 "The actual value may be smaller if the actual length of source_datapipe is smaller than the limit."
             )
             return self.limit
+
+
+@functional_datapipe("limit")
+class LimiterIterDataPipe(IterDataPipe[T_co]):
+    """
+    Limits the number of samples that the source DataPipe can yield based the input as well as the number of
+    processes and workers if distribute- or multi-processing is enabled (functional name: ``limit``).
+
+    The number of samples that will be yielded equals ``num_take // (num_processes * worker_info.num_workers)``.
+
+    Args:
+        source_datapipe: IterDataPipe with samples of data that will be yielded
+        num_take: the number of samples that you wish to yield across all processes and workers, should be less than
+            or equal to the ``len(source_datapipe)``
+
+    Example:
+        >>> source_dp = IterableWrapper(range(20))
+        >>> limiter_dp = source_dp.limit(5)
+        >>> list(limiter_dp)
+        [0, 1, 2, 3, 4]
+    """
+
+    def __init__(self, source_datapipe: IterDataPipe[T_co], num_take: int) -> None:
```

Comment:
The name `num_take` was somewhat tied to the name `TakerDataPipe`. Is this still expressive enough?

Reply:
If we're going for `LimiterIterDataPipe`, `limit` might be adequate

Reply:
Might be a little awkward though in the functional interface

```py
dp.limit(limit=...)`
```